frontend/src/components/placementVisuals/PlacementSamplesTable.tsx

```diff
@@ -21,113 +19,147 @@ interface PlacementSample {
+                if (arrayComparison < 0) orderA -= MAX / 4
+                if (arrayComparison > 0) orderB -= MAX / 4
+            }
+
+            if (a.name < b.name) orderA -= MAX / 8
+            if (a.name > b.name) orderB -= MAX / 8
+
+            if (a.containerName < b.containerName) orderA -= MAX / 16
+            if (a.containerName > b.containerName) orderB -= MAX / 16
+
+            if (a.projectName < b.projectName) orderA -= MAX / 32
+            if (a.projectName > b.projectName) orderB -= MAX / 32
+
+            return orderA - orderB
+        })
+        return samples
+    }, [container, samplesByID])
 
     const selectedRowKeys = useMemo(
-        () => container ? samples.reduce((sampleIDs, s) => {
-            if (s.selected && s.id) {
-                sampleIDs.push(s.id)
-            }
-            return sampleIDs
-        }, [] as FMSId[]) : [],
-        [container, samples]
+        () => samples.filter((s) => s.selected && s.id).map(rowKey),
+        [samples]
     )
     const onChange: NonNullable<TableRowSelection<PlacementSample>['onChange']> = useCallback((keys, selectedRows, info) => {
         if (info.type === 'all') {
             dispatch(multiSelect({
                 type: 'all',
-                parentContainer: containerName,
+                parentContainer: parentContainerID,
                 context: {
-                    source: activeSourceContainer?.name
+                    source: activeSourceContainer
                 }
             }))
         }
-    }, [activeSourceContainer?.name, containerName, dispatch])
+    }, [activeSourceContainer, parentContainerID, dispatch])
     const onSelect: SelectionSelectFn<PlacementSample> = useCallback((sample, selected) => {
-        dispatch(multiSelect({
-            type: 'sample-ids',
-            parentContainer: containerName,
-            sampleIDs: [sample.id],
-            forcedSelectedValue: selected,
-            context: {
-                source: activeSourceContainer?.name
-            }
-        }))
-    }, [activeSourceContainer?.name, containerName, dispatch])
+        if (!activeSourceContainer) return
+        if (parentContainerID.name === null) {
+            dispatch(multiSelect({
+                forcedSelectedValue: selected,
+                context: {
+                    source: activeSourceContainer
+                },
+                type: 'sample-ids',
+                parentContainer: parentContainerID,
+                samples: [sample],
+            }))
+        } else {
+            if (!sample.coordinates) return
+            dispatch(multiSelect({
+                forcedSelectedValue: selected,
+                context: {
+                    source: activeSourceContainer
+                },
+                type: 'samples-placements',
+                parentContainer: parentContainerID,
+                samples: [{ sample, cell: { fromContainer: parentContainerID, coordinates: sample.coordinates } }],
```

Comment:
Exposing a different format for samples here is risky and forcing you to track and test it all over the place ... you should have a uniform format (at least list sample in an object so you do not need to test it all the time) and test it within your placement objects as much as possible.